src/frontend/js/project-service.js

```diff
@@ -36,6 +37,17 @@ angular.module('seashell-projects', ['seashell-websocket', 'marmoset-bindings'])
       var PROJ_FILE_LIST_URL_TEMPLATE = CGI_URL + "skeleton_file_list.rkt?template=%s";
       var PROJ_WHITE_LIST_URL = CGI_URL + "project_whitelist.cgi";
       var USER_WHITE_LIST_URL = CGI_URL + "user_whitelist.cgi";
+
+      localfiles.init().then(function () { 
+        ws.register_callback("connected", function () {
+          settings.load() // need to load so that we can determine if we are forceOffline
+            .then(function () { 
+              if (ws.isOnline()) {
+                localfiles.syncOfflineChanges(self.syncOfflineChanges);          
```

Comment:
We should also syncOfflineChanges when coming back online; see above comment for note on handling the syncing in the websocket service itself.